### ISSUE TITLE
CI/DOC: pyarrow ImportError when using environment.yml to build the docs

### ISSUE THREAD

**Issue by datapythonista** [pandas-dev/pandas#26656]
In #26648, when we tried to use `environment.yml` to build the docs in azure instead of `ci/deps/travis-36-docs.yaml`, we found that `pyarrow` couldn't be imported, and it failed with the next error :

```
ImportError: /home/vsts/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pyarrow/lib.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZNK5arrow6Status8ToStringB5cxx11Ev
```

There are few differences between the created environments, like Python 3.6 vs 3.7, arrow-cpp being installed from defaults vs conda-forge... But it's unclear what's the exact problem or what is causing it.

Ideally, we'd like to use `environment.yml` to build the docs, so the environment is exactly the same as the local installations we use.

What we need to do is:

- Identify what is causing the error
- Implement the required changes so `pyarrow` can be imported
- Replace `ci/deps/travis-36-docs.yaml` by `environment.yml` in the doc builds in Travis and Azure.

**Comment by rjsee** [comment]
Could be inconsistent gcc settings, see 
https://issues.apache.org/jira/browse/ARROW-4809
and similarly
https://github.com/ray-project/ray/issues/2483
OTOH might be something else in the environment file upstream that's then leading to this conflict

**Comment by jreback** [comment]
is this still an issue? @minggli closed #26670 as a patch for this

**Comment by datapythonista** [comment]
Looks like the problem is not happening anymore. I guess it was a bug on Anaconda's side serving `arrow-cpp` from the `conda-forge` channels instead of the defaults conda one.

Checked in the logs of: #26829
